src/org/sasylf/refactor/RenameTheoremWizard.java

```diff
@@ -0,0 +1,229 @@
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.FindReplaceDocumentAdapter;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.IRegion;
+import org.eclipse.jface.text.Region;
+import org.eclipse.jface.wizard.IWizard;
+import org.eclipse.jface.wizard.Wizard;
+import org.eclipse.text.edits.MalformedTreeException;
+import org.eclipse.text.edits.MultiTextEdit;
+import org.eclipse.text.edits.ReplaceEdit;
+import org.eclipse.text.edits.TextEdit;
+import org.eclipse.text.edits.TextEditProcessor;
+import org.eclipse.ui.PlatformUI;
+import org.sasylf.IDEProof;
+import org.sasylf.project.ProjectModuleFinder;
+import org.sasylf.project.ProofBuilder;
+import org.sasylf.util.EclipseUtil;
+
+import edu.cmu.cs.sasylf.ast.CompUnit;
+import edu.cmu.cs.sasylf.ast.Node;
+import edu.cmu.cs.sasylf.ast.QualName;
+import edu.cmu.cs.sasylf.ast.RuleLike;
+import edu.cmu.cs.sasylf.ast.Theorem;
+import edu.cmu.cs.sasylf.module.ModuleId;
+import edu.cmu.cs.sasylf.util.Location;
+
+public class RenameTheoremWizard extends Wizard implements IWizard {
+	private RenameTheoremWizardPage page;
+	private RefactoringContext context;
+	private Map<IDocument, TextEdit> changes;
+
+	public RenameTheoremWizard(RefactoringContext context) {
+		super();
+		setNeedsProgressMonitor(true);
+		this.context = context;
+		changes = new HashMap<IDocument, TextEdit>();
+	}
+
+	@Override
+	public void addPages() {
+		page = new RenameTheoremWizardPage(context);
+		addPage(page);
+	}
+
+	/**
+	 * This method is called when 'Finish' button is pressed in
+	 * the wizard. We will create an operation and run it
+	 * using wizard as execution context.
+	 */
+	@Override
+	public boolean performFinish() {
+		try {
+			getOriginalFileChange();
+			collectReferencedChanges();
+			makeChanges();
+			PlatformUI.getWorkbench().saveAllEditors(false);	// save all changed editors
+			return true;
+		} catch (Exception e) {
+			e.printStackTrace();
+			return false;
+		}
+	}
+
+	private void getOriginalFileChange() throws BadLocationException {
+		IDocument doc = context.getProofDocument();
+		MultiTextEdit edit = new MultiTextEdit();
+		IRegion moduleLoc = null;
+		String newName = page.getTheoremName();
+
+		Map<String, ? super RuleLike> map = new HashMap<>();
+		context.getCompUnit().collectRuleLike(map);
+		Theorem theorem = (Theorem)map.get(context.getOldName());
+
+		moduleLoc = getModuleLocation(doc, theorem);
+
+		int offset = moduleLoc.getOffset();
+		int length = moduleLoc.getLength();
+
+		if (doc.get(offset, length).equals(newName)) {
+			// no change needed
+			return;
+		}
+		System.out.println("Replacing " + doc.get(offset, length) + " with " + newName);
+		edit.addChild(new ReplaceEdit(offset, length, newName));
+
+		List<QualName> qualNames = new ArrayList<>();
+		Consumer<QualName> consumer = name -> {			
+			if (name.getLastSegment().equals(context.getOldName())) {
+				QualName source = name.getSource();
+				if (source == null) {
```

Comment:
If you checked resolution instead, could perhaps this method and the next be merged?